lib/api/readable.js

```diff
@@ -147,38 +147,25 @@ module.exports = class BodyReadable extends Readable {
   async dump (opts) {
     let limit = opts && Number.isFinite(opts.limit) ? opts.limit : 262144
     const signal = opts && opts.signal
-    function throwIfAborted () {
-      if (!signal) { return }
-      if (signal.throwIfAborted) {
-        signal.throwIfAborted()
-      } else {
-        if (signal.aborted) {
-          // DOMException not available < v17.0.0
-          const err = new Error('The operation was aborted')
-          err.name = 'AbortError'
-          throw err
-        }
-      }
-    }
     if (signal) {
       if (typeof signal !== 'object' || !('aborted' in signal)) {
         throw new InvalidArgumentError('signal must be an AbortSignal')
       }
-      throwIfAborted()
+      util.throwIfAborted(signal)
       signal.addEventListener('abort', () => {
         this.destroy()
-      })
+      }, { once: true })
```

Comment:
this will still leak one we completed without abort

Reply:
add a finally clause to remove it

Reply:
Ah sorry, completely distracted need more chai sigh, anyway have updated